code/game/objects/items/weapons/storage/boxes.dm

```diff
@@ -981,3 +981,304 @@
 	new /obj/item/weapon/reagent_containers/glass/beaker/erlenmeyer/sodawater(src)
 	new /obj/item/weapon/reagent_containers/glass/beaker/large/erlenmeyer(src)
 	new /obj/item/weapon/paper/diy_soda(src)
+
+
+//Smart boxes.
+/obj/item/weapon/storage/box/smartbox
+	name = "Smart-box"
+	desc = "A one-use box that leaves no trash or cardboard behind."
+	foldable = null
+	storage_slots = BOX_SPACE
+	var/one_way = 0 //For one way boxes, you can take out but not put in. Could be moved to /box.
+
+
+/obj/item/weapon/storage/box/smartbox/remove_from_storage(obj/item/W as obj, atom/new_location, var/force = 0, var/refresh = 1)
+	. = ..()
+	if(contents.len <= 0) //If this is the last item, kill the box.
+		to_chat(new_location,"<span class='notice'>The [src] fizzles away into a glittering dust.</span>") //Should probably typecheck new_location as a mob?
+		qdel(src)
+		return
+	return . //remove_from_storage returns 1 or 0 :)
+
+/obj/item/weapon/storage/box/smartbox/attackby(obj/item/W, mob/user)
+	if(one_way)
+		to_chat(user, "<span class='warning'>\the [src] only lets items leave it!</span>") //Couldn't think of something better to explain the oneway interaction in-game.
+		return
+	else
+		..()
+
+//Clothing-boxes.
+/obj/item/weapon/storage/box/smartbox/clothing_box
+	name = "box"
+	desc = "A smart-box style box for clothing, convenient for distributing clothes."
+	icon_state = "clothing_box"
+	foldable = null
+	storage_slots = BOX_SPACE
+	one_way = 1
+
+//Every clothing box will base its label overlay off of the first object in its contents. Keep that in mind when making a new clothing box.
+/obj/item/weapon/storage/box/smartbox/clothing_box/New()
+    ..()
+    if(contents.len)
+        var/mutable_appearance/M = new(contents[1])
+        M.layer = FLOAT_LAYER
+        M.plane = FLOAT_PLANE
+        M.transform *= 0.5
+        overlays += M
+
+/obj/item/weapon/storage/box/smartbox/clothing_box/chickensuit
+	name = "Chicken suit box"
+
+/obj/item/weapon/storage/box/smartbox/clothing_box/chickensuit/New()
+	new	/obj/item/clothing/head/chicken(src)
```

Comment:
weird tab

Reply:
Looks like a github thing, its fine on my end. 
![image](https://user-images.githubusercontent.com/29109612/68133378-dcfc5100-fee5-11e9-9a07-ddd19bdd3868.png)

Reply:
It's because it's a tab
Why is it a tab?